utils/grpc/setup_server.go

```diff
@@ -0,0 +1,87 @@
+package grpc
+
+import (
+	"log"
+	"net"
+
+	"github.com/tron-us/go-btfs-common/protos/shared"
+	"github.com/tron-us/go-btfs-common/protos/status"
+
+	"github.com/tron-us/go-common/v2/constant"
+	"github.com/tron-us/go-common/v2/middleware"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/reflection"
+)
+
+type GrpcServer struct {
+	server       *grpc.Server
+	healthServer *health.Server
+	serverName   string
+	lis          net.Listener
+	dBURL        string
+	rDURL        string
+}
+
+func (s *GrpcServer) GrpcStatusServer(port string, dBURL string, rDURL string, statusServer status.StatusServiceServer) *GrpcServer {
+	s.dBURL = dBURL
+	s.rDURL = rDURL
+
+	lis, err := net.Listen("tcp", port)
+	if err != nil {
+		log.Panic(constant.RPCListenError, zap.Error(err))
+	}
+	s.lis = lis
+	s.CreateServer("status-server").
+		CreateHealthServer().
+		RegisterServer(statusServer).
+		RegisterHealthServer().
+		WithReflection().
+		AcceptConnection()
```

Comment:
should be more generic, and let services pass in the customize values

Reply:
ok